src/lib.rs

```diff
@@ -261,3 +262,185 @@ pub async fn webhook(
         Ok(true)
     }
 }
+
+const PREF_ALLOW_PING_AFTER_DAYS: i32 = 20;
+const PREF_MAX_ASSIGNED_PRS: i32 = 5;
+
+#[derive(Debug, Serialize)]
+pub struct ReviewCapacityUser {
+    pub username: String,
+    pub id: uuid::Uuid,
+    pub checksum: String,
+    pub user_id: i64,
+    pub assigned_prs: Vec<i32>,
+    pub num_assigned_prs: Option<i32>,
+    pub max_assigned_prs: Option<i32>,
+    pub pto_date_start: Option<chrono::NaiveDate>,
+    pub pto_date_end: Option<chrono::NaiveDate>,
+    pub active: bool,
+    pub allow_ping_after_days: Option<i32>,
+    pub publish_prefs: bool,
+}
+
+impl ReviewCapacityUser {
+    pub(crate) fn is_available(&self) -> bool {
+        let today = chrono::Utc::today().naive_utc();
+        let is_available = (self.pto_date_end.is_some() && self.pto_date_start.is_some())
+            && (self.pto_date_end.unwrap() < today || self.pto_date_start.unwrap() > today);
+        self.active && is_available
+    }
+
+    // thin compat. layer, will be removed after stabilizing the new PR assignment
+    fn phony(username: String) -> Self {
+        Self {
+            username,
+            id: uuid::Uuid::new_v4(),
+            user_id: -1,
+            checksum: String::new(),
+            assigned_prs: vec![],
+            num_assigned_prs: None,
+            max_assigned_prs: None,
+            pto_date_start: None,
+            pto_date_end: None,
+            active: false,
+            allow_ping_after_days: None,
+            publish_prefs: false,
+        }
+    }
+}
+
+impl From<HashMap<String, String>> for ReviewCapacityUser {
```

Comment:
this `From` parses the form from the backoffice. I'm not too happy about this code though ...